tg-x/network-architecture.md

```diff
@@ -0,0 +1,234 @@
+---
+title: Anoma Network Architecture
+category: distributed-systems
+publish_date: 2024-07-07
+image: media/network-architecture.jpg
+excerpt: A brief introduction to the network architecture of Anoma.
+---
+
+## Introduction
+
+Anoma is researching and developing a set of network protocols
+that facilitate message passing in a distributed system.
+
+The network protocol stack is designed as a standalone component
+generically useful for building distributed systems.
+
+This article serves as a brief introduction and high-level overview of the network architecture.
+
+## Node architecture
+
+The system is based on the *actor model*,
+where each node runs a set of processes, or actors.
+We call these *engines* in our design.
+Engines have state and communicate with each other via asynchronous message passing:.
+An engine may send and respond to messages,
+react to timer events, and spawn other engines.
+
+Each node has a cryptographic identity that uniquely identifies it in the network.
+A node consists of a collection of running engines.
+Local engines communicate with each other directly,
```

Comment:
directly 
→ unencrypted message passing  

or do you mean

directly
→ directly (in principle via shared memory)

not sure how to avoid the idea that we suddenly do not do message passing any more